modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/integration/FunctionCommercePaymentIntegration.java

```diff
@@ -216,18 +258,47 @@ private JSONObject _getPayloadJSONObject(
 
 		JSONObject typeSettingsJSONObject = _jsonFactory.createJSONObject();
 
-		typeSettingsUnicodeProperties.forEach(
-			(key, value) -> typeSettingsJSONObject.put(key, value));
+		typeSettingsUnicodeProperties.forEach(typeSettingsJSONObject::put);
+
+		JSONObject commercePaymentEntryJSONObject =
+			_jsonFactory.createJSONObject(
+				_jsonFactory.looseSerializeDeep(commercePaymentEntry));
+
+		String className = _portal.getClassName(
+			commercePaymentEntry.getClassNameId());
+
+		commercePaymentEntryJSONObject.put(
+			"callbackURL",
```

Comment:
why are we adding these to the JSONObject instead of updating the CommercePaymentEntry and then serialize that one?
We are actually not storing these info in the CommercePaymentEntry this way.